tests/arp/test_stress_arp.py

```diff
@@ -215,3 +218,62 @@ def test_ipv6_nd(duthost, ptfhost, config_facts, tbinfo, ip_and_intf_info,
                     raise e
             # Wait for 10 seconds before starting next loop
             time.sleep(10)
+
+
+def add_nd_nonstop(ptfadapter, ip_and_intf_info, ptf_intf_index, nd_available, stop_event):
+    while not stop_event.is_set():
+        entry = random.randrange(0, nd_available)
+        nd_entry_mac = IntToMac(MacToInt(ARP_SRC_MAC) + entry)
+        fake_src_addr = generate_global_addr(nd_entry_mac)
+        ns_pkt = ipv6_packets_for_test(ip_and_intf_info, nd_entry_mac, fake_src_addr)
+        testutils.send_packet(ptfadapter, ptf_intf_index, ns_pkt)
+
+
+def test_ipv6_nd_incomplete(duthost, ptfhost, config_facts, tbinfo, ip_and_intf_info,
+                            ptfadapter, get_function_completeness_level, proxy_arp_enabled,
+                            skip_traffic_test):    # noqa F811
+    _, _, ptf_intf_ipv6_addr, _, ptf_intf_index = ip_and_intf_info
+    ptf_intf_ipv6_addr = increment_ipv6_addr(ptf_intf_ipv6_addr)
+    pytest_require(proxy_arp_enabled, 'Proxy ARP not enabled for all VLANs')
+    pytest_require(ptf_intf_ipv6_addr is not None, 'No IPv6 VLAN address configured on device')
+
+    ipv6_available = get_crm_resources(duthost, "ipv6_neighbor", "available")
+    fdb_available = get_crm_resources(duthost, "fdb_entry", "available")
+    pytest_assert(ipv6_available > 0 and fdb_available > 0, "Entries have been filled")
+
+    nd_available = min(min(ipv6_available, fdb_available), ENTRIES_NUMBERS)
+
+    pytest_assert("[UNREPLIED]" not in duthost.command("sudo conntrack -f ipv6 -L dying")["stdout"],
+                  "unreplied icmpv6 requests ended up in the dying list before test is run")
+
+    try:
+        duthost.command("sudo ip6tables -I INPUT -p ipv6-icmp -j DROP --icmpv6-type neighbour-advertisement")
+        logger.info("drop incoming neighbour-advertisement packets with ip6tables")
+
+        clear_dut_arp_cache(duthost)
+
+        stop_event = threading.Event()
+        thread = InterruptableThread(
+            target=add_nd_nonstop,
+            args=(ptfadapter, ip_and_intf_info, ptf_intf_index, nd_available, stop_event),
+        )
+        thread.daemon = True
+        thread.start()
+        logger.info("started process to keep sending neighbour-solicitation from ptf to dut")
+
+        time.sleep(20)  # wait for incomplete state entry to accumulate
+
+        logger.info("neighbors in INCOMPLETE state: {}"
+                    .format(duthost.command("ip -6 neigh")["stdout"].count("INCOMPLETE")))
+
+        if not skip_traffic_test:
```

Comment:
why is the check for entries in dying list within the "if not skip_traffic_test" condition? Shouldn't we check for this regardless of skip_traffic_test

Reply:
removed.